sycl/test-e2e/XPTI/lit.local.cfg

```diff
@@ -0,0 +1,2 @@
+# Prepending instead of appending so that `%shared_lib` is expanded before `%s`.
+config.substitutions.insert(0, ("%build_collector", "%clangxx %s -DXPTI_COLLECTOR -DXPTI_CALLBACK_API_EXPORTS %xptifw_lib %shared_lib %fPIC %cxx_std_optionc++17 -o %t_collector.dll %if preview-mode %{-Wno-unused-command-line-argument%}"))
```

Comment:
Shouldn't `%t_collector.dll` be `%t_collector.so` on Linux and DLL only on Windows?

Reply:
Similar comment for tests using `%build_collector`

Reply:
I'm not really "changing" this. The usage of `.so`/`.dll` was somewhat inconsistent before. At least I'm making it uniform everywhere. Also note how `-shared/-fPIE` became uniform now.

Reply:
I see. Fine with me then.